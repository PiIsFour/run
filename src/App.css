.App {
	text-align: center;
	display: flex;
	flex-direction: column-reverse;
	align-items: center;
}

span {
	display: inline-block;
	margin: .5em;
}
