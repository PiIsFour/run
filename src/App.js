import React, { Component } from 'react'
import './App.css'

import {gpsStream} from './gps/gps'

function Pos ({pos}) {
	return <div>
		{Object.entries(pos).map(([key, value]) => <span key={key}>{key}: {value}</span>)}
	</div>
}

class App extends Component {
	constructor (props) {
		super(props)
		this.state = {pos: []}
	}
	render () {
		let north = 0
		let east = 0
		if (this.state.pos.length > 0) {
			const u = 40000000
			const first = this.state.pos[0]
			const last = this.state.pos[this.state.pos.length - 1]
			north = (last.lat - first.lat) * u / 360
			east = (last.lon - first.lon) * u / 360 * Math.cos(first.lat * Math.PI / 180)
		}
		return (
			<div className="App">
				{this.state.pos.map((p, index) => <Pos pos={p} key={index}/>)}
				<div><span>north: {north}m</span><span>east: {east}m</span></div>
			</div>
		)
	}
	componentDidMount () {
		const s = gpsStream(this.props.geolocation)
			.map(p => ({
				lat: p.coords.latitude,
				lon: p.coords.longitude,
				accuracy: p.coords.accuracy,
				date: new Date(p.timestamp).toLocaleString()
			}))
		s.map(p => {
			this.setState({pos: [...this.state.pos, p]})
			s.map(next => ({
				...next
			}))
				.map(p => this.setState({pos: [...this.state.pos, p]}))
			return 0
		})
	}
}

export default App
