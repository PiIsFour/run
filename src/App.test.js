import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import enzymeToJson from 'enzyme-to-json'

import App from './App'

describe('app', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<App geolocation={{watchPosition: () => {}}} />, div)
		ReactDOM.unmountComponentAtNode(div)
	})

	it('snapshot matches', () => {
		const wrapper = shallow(<App geolocation={{watchPosition: () => {}}} />)
		expect(enzymeToJson(wrapper)).toMatchSnapshot()
	})
})
