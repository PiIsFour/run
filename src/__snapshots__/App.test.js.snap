// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot matches 1`] = `
<div
  className="App"
>
  <div>
    <span>
      north: 
      0
      m
    </span>
    <span>
      east: 
      0
      m
    </span>
  </div>
</div>
`;
